src/v/kafka/protocol/schemata/generator.py

```diff
@@ -1335,21 +1335,28 @@ class response;
 {%- endif %}
 {%- endmacro %}
 
-{% macro conditional_tag_encode(tdef, vec) %}
+{% macro conditional_tag_encode(tdef, vec, obj = "") %}
+{%- if obj %}
+{%- set fname = obj + "." + tdef.name %}
+{%- else %}
+{%- set fname = tdef.name %}
+{%- endif %}
 {%- if tdef.nullable() %}
-if ({{ tdef.name }}) {
+if ({{ fname }}) {
     {{ vec }}.push_back({{ tdef.tag() }});
 }
 {%- elif tdef.is_array %}
-if (!{{ tdef.name }}.empty()) {
+if (!{{ fname }}.empty()) {
     {{ vec }}.push_back({{ tdef.tag() }});
 }
 {%- elif tdef.default_value() != "" %}
-if ({{ tdef.name }} != {{ tdef.default_value() }}) {
+if ({{ fname }} != {{ tdef.default_value() }}) {
```

Comment:
So the logic was already there it just wasn't working?

Reply:
This logic only applied if the schemata set a default value.  This worked fine actually for `api_versions_response` that does have a default value for `FinalizedFeaturesEpoch`.  This addressed a bug for if the tag existed in an object nested within the message.